src/shadow-catcher.ts

```diff
@@ -0,0 +1,132 @@
+import {
+    BLEND_NORMAL,
+    SHADOW_VSM16 as SHADOW_TYPE,
+    SHADOWUPDATE_REALTIME as SHADOWUPDATE,
```

Comment:
Maybe it's just me, but I found this use of `as` confusing when I read the light component creation code.